simulator.py

```diff
@@ -0,0 +1,41 @@
+import numpy
+
+
+class Simulator(object):
+    def __init__(self, state_space, action_space, discount):
+        self.state_space = state_space
+        self.action_space = action_space
+        self.discount = discount
+        self.max_iterations = 1000
+        self.tolerance = 0.01
+
+    def find_best_policy_for_state(self, state, value_func):
+        best_action = None
+        best_payoff = float('-inf')
+        for a, payoff in self.action_space.get_actions(state):
+            for next_state, prob in self.action_space.transition_probabilities(a, state):
+                payoff += value_func[next_state] * prob * self.discount
```

Comment:
I'm not sure what you mean by accumulate? If it were me, I'd write

```
for a, payoff in self.action_space.get_actions(state):
    acc = payoff
    for next_state, prob in self.action_space.transition_probabilities(a, state):
        acc += value_func[next_state] * prob * self.discount
```

Maybe rename `acc` and `payoff` to correspond to what they're called in the algorithm, like `total_payoff` and `initial_payoff`